localstack-core/localstack/services/apigateway/next_gen/execute_api/router.py

```diff
@@ -0,0 +1,114 @@
+import logging
+
+from rolo import Request, Router
+from rolo.dispatcher import Handler
+from werkzeug.routing import Rule
+
+from localstack.constants import AWS_REGION_US_EAST_1, DEFAULT_AWS_ACCOUNT_ID
+from localstack.http import Response
+from localstack.services.apigateway.models import ApiGatewayStore, apigateway_stores
+from localstack.services.edge import ROUTER
+
+from .context import RestApiInvocationContext
+from .gateway import RestApiGateway
+
+LOG = logging.getLogger(__name__)
+
+
+# TODO: subclass from -ext, and this handler will do the dispatching between v1 and v2 Gateways
+class ApiGatewayHandler:
+    def __init__(self, rest_gateway: RestApiGateway = None, store: ApiGatewayStore = None):
+        self.rest_gateway = rest_gateway or RestApiGateway()
+        # we only access CrossAccount attributes in the handler, so we use a global store in default account and region
+        self._global_store = (
+            store or apigateway_stores[DEFAULT_AWS_ACCOUNT_ID][AWS_REGION_US_EAST_1]
+        )
+
+    def __call__(self, request: Request, **kwargs) -> Response:
+        # api_id can be cased because of custom-tag id
+        api_id, stage = kwargs.get("api_id", "").lower(), kwargs.get("stage")
```

Comment:
Awesome! This means we are finally using `werkzeug`'s request parsing directly from the route matching! 🚀 🤩